lib/money/money.rb

```diff
@@ -337,6 +341,9 @@ def initialize( obj, currency = Money.default_currency, options = {})
     @currency ||= Money.default_currency
     @bank       = obj.respond_to?(:bank) ? obj.bank : options[:bank]
     @bank     ||= Money.default_bank
+    @infinite_precision = obj.respond_to?(:infinite_precision?) ? obj.infinite_precision? : options[:infinite_precision]
+    @infinite_precision = self.class.default_infinite_precision if @infinite_precision.nil?
+    @infinite_precision = !!@infinite_precision # ensure we have a boolean
```

Comment:
nit: Can you please move this logic ^^^ into a private method to reduce clutter in the initializer?

Reply:
sure